File: app/assets/javascripts/angular-controllers/UsersController.js
angular.module('snapApp')
    .controller('UsersController', UsersController);

UsersController.$inject = ['$http', '$routeParams', '$window'];

function UsersController($http, $routeParams, $window){
  var self = this;

  self.params = $routeParams

 self.createUser = function(){
      var newUser = {
        name: self.name,
        email: self.email,
        password: self.password,
        password_confirmation: self.password_confirmation,
        address_line_1: self.address_line_1,
        city: self.city,
        state: self.state,
        zipcode: self.zipcode
      };

      console.log(newUser)

      $http.post('/api/users', {user: newUser})
          .success(function(data){
              console.log('successfully created user');
              $window.location.href = ("#/cameras")
          })
          .error(function(data){
              console.log("Could not create user account. ")
              console.dir(data)
          });
 }; //end of createUser method


self.showUser = function(){
    var url = "/api/users/" + self.params.id;

    console.log(self.params);
    console.log(url)

    $http.get(url)
          .success(function(data){
                console.log("successfull show request")
                self.currentUser = data;
          })
          .error(function(data){
              console.log("Error!")

          })
} //end of showDog function

}; // end of DogsController

File: app/controllers/api/users_controller.rb
module API

    class UsersController < ApplicationController

        before_action :restrict_access, only: [:update,:destroy]

        # skip_before_action  :verify_authenticity_token

        def index
           @users = User.all

           render json: @users
        end

        def show
            @user = User.find(params[:id])

            render json: @user
        end

         def create
            @user = User.new(params.require(:user).permit(:name, :email, :password, :password_confirmation, :address_line_1, :city, :state, :zipcode))

            if @user.save
                render json: @user, status: 201
              else
                render json: {errors: @user.errors}, status: 422
            end

        end #end of create method

          private
          def restrict_access
              token = User.find_by(token: params[:token])
            render json: {error:"You need to be logged in to access this"}, status: 401 unless token
            end 

    end #end of UsersController

end #end of API module